apps/api/v2/src/ee/bookings/2024-08-13/guards/booking-pbac.guard.ts

```diff
@@ -0,0 +1,54 @@
+import { ApiAuthGuardUser } from "@/modules/auth/strategies/api-auth/api-auth.strategy";
+import { PrismaReadService } from "@/modules/prisma/prisma-read.service";
+import {
+  Injectable,
+  CanActivate,
+  ExecutionContext,
+  ForbiddenException,
+  UnauthorizedException,
+  BadRequestException,
+} from "@nestjs/common";
+import { Request } from "express";
+
+import { BookingAccessService } from "@calcom/platform-libraries";
+
+@Injectable()
+export class BookingPbacGuard implements CanActivate {
+  private bookingAccessService: BookingAccessService;
+
+  constructor(private readonly prismaReadService: PrismaReadService) {
+    this.bookingAccessService = new BookingAccessService(this.prismaReadService.prisma);
+  }
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    const request = context.switchToHttp().getRequest<Request & { pbacAuthorizedRequest?: boolean }>();
+    const user = request.user as ApiAuthGuardUser;
+    const bookingUid = request.params.bookingUid;
+
+    if (!user) {
+      throw new UnauthorizedException(
+        "BookingPbacGuard - the request does not have an authorized user provided"
```

Comment:
This feels out of spec with most APIs - need to throw a generic error imo (http error 403, no other data)

Reply:
> This feels out of spec with most APIs 

We are using this pattern in other API V2 services/classes as well. I'll remove the text here if we want